staging/src/k8s.io/apimachinery/pkg/util/version/version.go

```diff
@@ -215,6 +252,18 @@ func (v *Version) WithMinor(minor uint) *Version {
 	return &result
 }
 
+// SubtractMinor returns the version diff minor versions back, with the same major and patch.
+// If diff >= current minor, the minor would be 0.
+func (v *Version) SubtractMinor(diff uint) *Version {
+	result := *v
+	var minor uint
+	if diff < v.Minor() {
+		minor = v.Minor() - diff
+	}
+	result.components = []uint{v.Major(), minor, v.Patch()}
```

Comment:
Why is it better to keep an invalid patch version?  could we set it to maxInt16?  That would still parse, but we're unlikely to ever ship one that high.  Otherwise we might think we're actually a version with a CVE for instance.